src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxIndexWriteOperation.cs

```diff
@@ -52,6 +49,17 @@ public CoraxIndexWriteOperation(Index index, Transaction writeTransaction, Corax
             {
                 throw new IndexWriteException(e);
             }
+            
+            if (index.Definition.HasDynamicFields)
+            {
+                _dynamicFieldsBuilder = new IndexFieldsMappingBuilder(true, true)
+                    .AddDefaultAnalyzer(knownFields.DefaultAnalyzer)
+                    .AddExactAnalyzer(knownFields.ExactAnalyzer)
+                    .AddSearchAnalyzer(knownFields.SearchAnalyzer);
```

Comment:
If we fail on any `Add...` then we won't release the context allocated by `new IndexFieldsMappingBuilder(true, true)`